### DISCUSSION TITLE
Issues when Running HyMAP on University Servers

### DISCUSSION THREAD

**Discussion by osbornesci** [NASA-LIS/LISF#981] [General]
We're experiencing issues when running LIS with HyMAP within servers for two different universities. LIS runs are successful only when HyMAP routing is not included.

It's possible issues are memory or parameter related. So far, we've tried reducing the domain size, running with 4 instead of 9 processors, and rechecking HyMAP specifications in the config file, but issues remain.

Could there be a memory concern involved with HyMAP updates that leads to processes being killed? Is it possible that we’ll need to subset the HyMAP parameter files (more digging into the HyMAP source code to update the reader)? 

There may be something we're missing here. Please let us know if you have advice.

In case it's helpful, we're running the Noah-MP v3.6 LSM within the LIS v7.3.2 modeling framework.

Error message after running LIS with HyMAP using mpirun:
![ErrorMsg1](https://user-images.githubusercontent.com/99269620/153023449-c6fbe59e-598d-4365-824b-8dd99b5e0869.png)

**Comment by agetiran** [comment]
Hi @osbornesci,
could you please share the HyMAP options in the lis.input file, domain size and spatial resolution, and last lines in the lislog file? 
I would also suggest that you recompile LIS in debug mode so you can get more information on the crash.

Augusto

**Reply by osbornesci** [reply]
Hi @agetiran, 
Thank you. I'm attaching the information you requested. 
Please let me know if anything stands out.

We can also continue troubleshooting.
Thanks again.

[HyMAP_Error_UniversityServer.pdf](https://github.com/NASA-LIS/LISF/files/8026687/HyMAP_Error_UniversityServer.pdf)

**Comment by dmocko** [comment]
Hi @osbornesci and @agetiran,

Is there any update with this issue?

I don't see anything obvious in the attachment that could be causing this.

I wonder if the "HYMAP router" is no longer working in parallel mode in this version.  Have you tried "HYMAP2 router"?  There are a few more parameters to add in LDT, and you should check the LIS Users' Guide for the HYMAP2 router config entries.

David

**Reply by osbornesci** [reply]
Thank you, @dmocko @agetiran. I'm returning to this discussion to provide an update. We have since switched over to using LIS v4, with Noah-MP v4.0 and HyMAP2. With the current setup, it looks like we're getting closer to a successful run on Discover (having an issue with writing streamflow output, here: https://github.com/NASA-LIS/LISF/discussions/1109). 

Once this is successful, I will run the same case on the two university servers (one in Maryland, the other in Ecuador) and repeat to see if we run into concerns there. Thank you again for your help - It is appreciated.

**Reply by PrateekSharma99** [reply]
Any updates on this issue @osbornesci 
Thankyou
